root/static/scripts/edit/URLCleanup.js

```diff
@@ -321,12 +322,13 @@ const CLEANUPS = {
     type: LINK_TYPES.discogs,
     clean: function (url) {
       url = url.replace(/\/viewimages\?release=([0-9]*)/, "/release/$1");
+      url = url.search(/\/composition\//) != -1 ? url.replace(/\/composition.*/, /\/composition([^-])+-[^-]+-[^-]+-[^-]+-[^-]+/.exec(url)[0]) : url;
```

Comment:
The calls to `search`, `exec`, and the ternary operator are not needed. KISS!

Reply:
When I don't use the ternary operator to check whether the URL is that of a composition, I get an error with accessing the element at index 0.
Also, without using ".exec", the URL gets replaced with the actual expression itself.
Is there a way that I can avoid these?

Reply:
Yes, use a single `replace` on the model of the line above.

Reply:
Additional hint: unlike your regexp, the above line is using _capture group_.

Reply:
Okay, I think I've sorted it out 👍